libutils/logging.h

```diff
@@ -141,10 +141,16 @@ void LogToSystemLog(const char *msg, LogLevel level);
  *          log level argument and added to the structured log message.
  * @note Only the formatted string from the pair with the "MESSAGE" key is
  *       logged on platforms that does not support structured logging.
+ *
+ *       The function takes an int as the log level, although the deired type
+ *       is LogLevel. However, it is not safe to use an enum next to a
+ *       variable length argument list. Hence, we must use a type of a fixed
```

Comment:
it is not safe, as I understand from a bit of reading, because the underlying type of enums is undefined/implementation specific and will be promoted to int when passed. Not sure if you want to add a few more details or a reference to a specification here.

Reply:
Added a bit more details. It's quite confusing, but so is variadic arguments in general. It's more of a hack